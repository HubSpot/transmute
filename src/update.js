import _get from './internal/_get';
import _set from './internal/_set';
import curry from './curry';

function update(key, updater, subject) {
  const value = _get(key, subject);
  return _set(updater(value), key, subject);
}

export default curry(update);
