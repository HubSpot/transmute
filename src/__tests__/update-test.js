import { Map } from 'immutable';
import update from '../update';

describe('transmute/update', () => {
  const incCount = update('count', n => n + 1);

  it('updates a Map', () => {
    expect(incCount(Map({ count: 1 }))).toMatchSnapshot();
  });

  it('updates an object', () => {
    expect(incCount({ count: 1 })).toMatchSnapshot();
  });
});
