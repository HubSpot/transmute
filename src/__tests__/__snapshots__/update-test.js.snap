// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/update updates a Map 1`] = `
Immutable.Map {
  "count": 2,
}
`;

exports[`transmute/update updates an object 1`] = `
Object {
  "count": 2,
}
`;
